x/poolmanager/keeper.go

```diff
@@ -208,3 +221,65 @@ func (k *Keeper) SetStakingKeeper(stakingKeeper types.StakingKeeper) {
 func (k *Keeper) SetProtorevKeeper(protorevKeeper types.ProtorevKeeper) {
 	k.protorevKeeper = protorevKeeper
 }
+
+// SetWasmKeeper sets wasm keeper
+func (k *Keeper) SetWasmKeeper(wasmKeeper types.WasmKeeper) {
+	k.wasmKeeper = wasmKeeper
+}
+
+// BeginBlock sets the poolmanager caches if they are empty
+func (k *Keeper) BeginBlock(ctx sdk.Context) {
+	// Set the caches if they are empty.
+	if len(k.cachedTakerFeeShareAgreementMap) == 0 || len(k.cachedRegisteredAlloyPoolByAlloyDenomMap) == 0 || len(k.cachedRegisteredAlloyedPoolIdArray) == 0 {
+		err := k.setTakerFeeShareAgreementsMapCached(ctx)
+		if err != nil {
+			ctx.Logger().Error(fmt.Errorf("%w", types.ErrSetTakerFeeShareAgreementsMapCached).Error())
+		}
+		err = k.setAllRegisteredAlloyedPoolsByDenomCached(ctx)
+		if err != nil {
+			ctx.Logger().Error(fmt.Errorf("%w", types.ErrSetAllRegisteredAlloyedPoolsByDenomCached).Error())
+		}
+		err = k.setAllRegisteredAlloyedPoolIdArrayCached(ctx)
+		if err != nil {
+			ctx.Logger().Error(fmt.Errorf("%w", types.ErrSetAllRegisteredAlloyedPoolsIdArrayCached).Error())
+		}
+	}
```

Comment:
I don't this that is the case here, but please correct me if I am wrong.

Here, the only time in which these caches are empty is during the start up of the node. Once the node has started up and runs the first BeginBlock of the poolmanager module, it will populate the caches. Every single subsequent BeginBlock, this logic will be a no-op.

Reply:
Right! Would appreciate a comment for clarity 🙏

Reply:
Added comment https://github.com/osmosis-labs/osmosis/pull/8310/commits/e6d85e0dbe2466b3ecd03b68ad29623837389d12